src/sentry/migrations/0049_auto_20200302_1955.py

```diff
@@ -0,0 +1,92 @@
+    #   SubPlan 1
+    #   -> Index Scan using sentry_organizationonboar_organization_id_47e98e05cae29cf3_uniq on sentry_organizationonboardingtask u0 (cost=0.42..4.45 rows=1 width=0) (never executed)
+    #   Index Cond: ((organization_id = sentry_project.organization_id) AND (task = 10))
+    #   Filter: ((status = 1) AND (project_id = sentry_project.id))
+    #   SubPlan 2
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_1 (cost=0.00..25585.33 rows=20264 width=16) (actual time=0.063..166.531 rows=20947 loops=1)
+    #   Filter: ((status = 1) AND (task = 10))
+    #   Rows Removed by Filter: 874039
+    #   SubPlan 3
+    #   -> Index Scan using sentry_organizationonboar_organization_id_47e98e05cae29cf3_uniq on sentry_organizationonboardingtask u0_2 (cost=0.42..4.45 rows=1 width=0) (never executed)
+    #   Index Cond: ((organization_id = sentry_project.organization_id) AND (task = 10))
+    #   Filter: ((status = 1) AND (project_id = sentry_project.id))
+    #   SubPlan 4
+    #   -> Seq Scan on sentry_organizationonboardingtask u0_3 (cost=0.00..25585.33 rows=20264 width=16) (actual time=0.081..203.465 rows=20947 loops=1)
+    #   Filter: ((status = 1) AND (task = 10))
+    #   Rows Removed by Filter: 874039
+    #   Planning time: 1.811 ms
+    #   Execution time: 913.912 ms
+    #
+    # A range QuerySet wrapper will be used to iterate this query.
+    target_projects = Project.objects.annotate(
+        alert_task_complete=Exists(
+            OrganizationOnboardingTask.objects.filter(
+                organization_id=OuterRef("organization_id"),
+                project_id=OuterRef("id"),
+                task=OnboardingTask.ALERT_RULE,
+                status=OnboardingTaskStatus.COMPLETE,
+            )
+        )
+    ).filter(alert_task_complete=False, date_added__lte=ALERT_RULE_ADDED_CUTOFF)
+
+    print (target_projects.query)
+    raise Exception("test")
+
+    for project in RangeQuerySetWrapperWithProgressBar(target_projects):
+        OrganizationOnboardingTask.objects.update_or_create(
+            organization_id=project.organization_id,
+            project_id=project.id,
+            task=OnboardingTask.ALERT_RULE,
+            defaults={"status": OnboardingTaskStatus.COMPLETE},
+        )
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = True
```

Comment:
Marked as dangerous. Though the query plan indicates the query isn't the slowest ever.

Reply:
It's more about the amount of time it's going to take to actually make those mutations. As well as that, the query itself will be run many times, not just once.